compaction/compaction_manager.cc

```diff
@@ -1962,7 +1962,7 @@ compaction_backlog_tracker::compaction_backlog_tracker(compaction_backlog_tracke
 compaction_backlog_tracker&
 compaction_backlog_tracker::operator=(compaction_backlog_tracker&& x) noexcept {
     if (this != &x) {
-        if (auto manager = std::exchange(_manager, x._manager)) {
+        if (auto manager = std::exchange(_manager, nullptr)) {
             manager->remove_backlog_tracker(this);
         }
```

Comment:
It's not clean if the correctness of compaction_backlog_manager move assignment depends on what its caller does.

Reply:
I agree.  I don't think this block is even needed since this is called only from `compaction_manager::register_backlog_tracker` and `cs.backlog_tracker` is either unregistered after it was initially constructed in `: backlog_tracker(t.get_compaction_strategy().make_backlog_tracker())` or it's already registered with the same manager.

Reply:
replaced with "compaction_backlog_tracker: operator=: do not change tracker registration" in v3 that just removes this block and documents that the function does not affect registration.